packages/test/test-service-load/src/leaderElection.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { ISignalMessage } from "@fluidframework/protocol-definitions";
+
+export class LeaderElection {
+    private readonly beatInEveryNSecs: number = 1000; // 1 secs
+    private readonly leaderWait: number = 5000; // 5 secs
+    private lastPinged: Date | undefined;
+
+    constructor(private readonly dataStoreRuntime: IFluidDataStoreRuntime) { }
+
+    public setupLeaderElection() {
+        this.dataStoreRuntime.on("signal", (signal: ISignalMessage) => this.handleSignal(signal));
+
+        const interval = setInterval(() => {
+            if (this.leaderId !== undefined && this.leaderId === this.dataStoreRuntime.clientId) {
+                this.dataStoreRuntime.submitSignal("leaderMessage", "leaderMessage");
+                this.lastPinged = new Date();
+            }else if(this.leaderId === undefined) {
+                this.dataStoreRuntime.logger.send(
+                    {
+                        category: "performance",
+                        eventName: "LeaderElection:Error",
+                        warning: "Leader is undefined.",
+                        leaderId: this.leaderId,
+                        clientId: this.dataStoreRuntime.clientId,
+                    });
+            }else {
+                const current = new Date();
+                if(this.lastPinged === undefined || current.getTime() - this.lastPinged?.getTime() > this.leaderWait) {
+                    this.dataStoreRuntime.logger.send(
```

Comment:
i'm worried this will be overly chatty. something simple could be to log only if lastPinged is not undefined, set it undefined in this condition, and initialize it to new Date. then we should only log once per lost leader, rather than once per missed ping, which over an multi hour stress run could be lots and lots of logs

Reply:
Same with the above Error telemetry, since the situation might never change, it will continue to generate the same telemetry.   Maybe we want to log start of error state and end of error state (which some count)?

Reply:
Updated